### PR TITLE
NO-JIRA: enforce termination policy

### PR DESCRIPTION
**Pull request by deads2k**
We're down to a few violations (first couple hundred fixed).  This updates the test to prevent new ones from being added.

### CODE REVIEW

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go [hunk 1]
```diff
@@ -79,7 +113,36 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
...
+			"containers[shared-resource-csi-driver-operator]",
+			"containers[vmware-vsphere-csi-driver-operator]",
+			"containers[vsphere-webhook]",
+			"pods/kubevirt-csi-node",
+			"pods/nutanix-csi-controller",
+			"pods/nutanix-csi-node",
+			"pods/ibm-vpc-block-csi-controller",
+			"pods/ibm-vpc-block-csi-node",
+			"pods/powervs-block-csi-driver-operator",
+		),
```
Comment (deads2k) [inline pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:136]: tidy list produced from https://search.dptools.openshift.org/?search=must+have+terminationMessagePolicy%3D&maxAge=48h&context=1&type=junit&name=4.16.*&excludeName=upgrade&maxMatches=5&maxBytes=20971520&groupBy=job

@jsafrane

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go [hunk 2]
```diff
@@ -79,7 +113,36 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
...
 	existingViolations := map[string]sets.String{
-		"namespace": sets.NewString("pods/<name> <containerType>[<containerName>]"),
+		"namespace": sets.NewString("<containerType>[<containerName>]"),
+		"openshift-catalogd": sets.NewString(
+			"pods/catalogd-controller-manager",
+		),
+		"openshift-cluster-api": sets.NewString(
+			"pods/capa-controller-manager",
+			"pods/capi-controller-manager",
+		),
```
Comment (deads2k) [inline pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:123]: @JoelSpeed

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go [hunk 3]
```diff
@@ -79,7 +113,36 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
...
+			"pods/nutanix-csi-node",
+			"pods/ibm-vpc-block-csi-controller",
+			"pods/ibm-vpc-block-csi-node",
+			"pods/powervs-block-csi-driver-operator",
+		),
+		"openshift-cloud-controller-manager": sets.NewString(
+			"pods/aws-cloud-controller-manager",
+		),
+		"openshift-machine-api": sets.NewString(
+			"containers[machine-ipam-controller]",
```
Comment (deads2k) [inline pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:141]: @JoelSpeed

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go [hunk 4]
```diff
@@ -79,7 +113,36 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
...
+			"containers[vsphere-webhook]",
+			"pods/kubevirt-csi-node",
+			"pods/nutanix-csi-controller",
+			"pods/nutanix-csi-node",
+			"pods/ibm-vpc-block-csi-controller",
+			"pods/ibm-vpc-block-csi-node",
+			"pods/powervs-block-csi-driver-operator",
+		),
+		"openshift-cloud-controller-manager": sets.NewString(
+			"pods/aws-cloud-controller-manager",
```
Comment (deads2k) [inline pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:138]: @JoelSpeed ?

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go [hunk 5]
```diff
@@ -79,7 +113,36 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
 
 	// existingViolations is the list of violations already present, don't add to it once we start enforcing
 	existingViolations := map[string]sets.String{
-		"namespace": sets.NewString("pods/<name> <containerType>[<containerName>]"),
+		"namespace": sets.NewString("<containerType>[<containerName>]"),
+		"openshift-catalogd": sets.NewString(
+			"pods/catalogd-controller-manager",
```
Comment (deads2k) [inline pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:118]: @joelanford I think you're waiting on a tag for these.

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go (hunk 1 above)
Comment (dobsonj) [reply pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:136]: I think this covers all of them except for kubevirt and nutanix:
https://github.com/openshift/cluster-storage-operator/pull/470
https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/231
https://github.com/openshift/ibm-vpc-block-csi-driver-operator/pull/114

File: pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go (hunk 3 above)
Comment (JoelSpeed) [reply pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go:141]: Redirect to @rvanderp3 , PTAL at https://github.com/openshift-splat-team/machine-ipam-controller/pull/7 for the fix

**Pull request merged**